### PR TITLE
feat💥: Improve channel permissions (see desc.)

### PR DESCRIPTION
**Pull request by leestarb**
## What type of pull request is this?

- [ ] Non-breaking code change
- [x] Breaking code change
- [x] Documentation change/addition

## Description

Some improvements for channel permissions related stuff
I'm open to any improvements

## Changes

- Implement `DictSerializationMixin` in `PermissionOverwrite`
- Rewrite `Channel.permission_overwrites` as list
- Make `Channel.permission_overwrites` public
- Documentate `Channel.permission_overwrites`
- Use docs for attr attributes in PermissionOverwrite
- Add a note for `id` attribute in PermissionOverwrite
- Remove useless `_original_permission_overwrites`
- Use `map`, `to_dict` for permissions in create channel methods
- Also use correct check for a empty list
- Add `Channel.permissions_for` helper method
- Add a note about it in `Member.channel_permissions`
- Refactor `Member.channel_permissions` to improve performance

## Checklist

- [x] I've formatted my code with [Black](https://black.readthedocs.io/en/stable/)
- [x] I've ensured my code works on `Python 3.10.x`
- [x] I've tested my code

### CODE REVIEW

File: dis_snek/models/discord/user.py
Comment (leestarb) [inline dis_snek/models/discord/user.py:389]: Does anyone know how to optimise this?

File: dis_snek/models/discord/channel.py
Comment (leestarb) [inline dis_snek/models/discord/channel.py]: Do not merge this PR unless its done!

Comment (leestarb) [reply dis_snek/models/discord/channel.py]: Done

**Review by LordOfPolls — APPROVED** [review]

**Pull request merged**
